File: src/main/java/com/bot/commands/voice/LockVolumeCommand.java
package com.bot.commands.voice;

import com.bot.commands.ModerationCommand;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;
import datadog.trace.api.Trace;
import org.springframework.stereotype.Component;

@Component
public class LockVolumeCommand extends ModerationCommand {

    public LockVolumeCommand() {
        this.name = "lockvolume";
        this.aliases = new String[]{"lvolume", "lvol", "lockv"};
        this.help = "Locks the volume for the playing stream. (Mod required)";
    }

    @Override
    @Trace(operationName = "executeCommand", resourceName = "LockVolume")
    protected void executeCommand(CommandEvent commandEvent) {
        VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
        if (handler == null) {
            commandEvent.replyWarning("I am not connected to a voice channel.");
            return;
        }

        String message = handler.toggleVolumeLock() ? "Volume is now locked" : "Volume is now unlocked";

        commandEvent.replySuccess(message);
    }

}

File: src/main/java/com/bot/interactions/commands/voice/LockVolumeCommand.kt
package com.bot.interactions.commands.voice

import com.bot.interactions.InteractionEvent
import com.bot.interactions.commands.ModerationCommand
import com.bot.voice.GuildVoiceProvider
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class LockVolumeCommand : ModerationCommand() {
    private val guildVoiceProvider = GuildVoiceProvider.getInstance()
    init {
        name = "lockvolume"
        aliases = arrayOf("lvolume", "lvol", "lockv")
        help = "Locks the volume for the playing stream. (Mod required)"
    }

    @Trace(operationName = "executeCommand", resourceName = "LockVolume")
    override fun executeCommand(commandEvent: CommandEvent) {
        val voiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild)
        if (!voiceConnection.isConnected()) {
            commandEvent.reply(commandEvent.client.warning + " I am not currently playing audio.")
            return
        }
        val message = if (voiceConnection.toggleVolumeLock()) "Volume is now locked" else "Volume is now unlocked"
        commandEvent.replySuccess(message)
    }

    override fun executeCommand(commandEvent: InteractionEvent?) {
        TODO("Not yet implemented")
    }
}